blockchain/blockchain.go

```diff
@@ -1056,13 +1069,18 @@ func (bc *BlockChain) writeStateTrie(block *types.Block, state *state.StateDB) e
 	trieDB := bc.stateCache.TrieDB()
 	trieDB.UpdateMetricNodes()
 
+	if bc.checkStartStateMigration(block, root) {
```

Comment:
After this, `trieDB.Commit()` will be called again.
Is it ok?

I think we can check the state migration prepare at `if isCommitTrieRequired(bc, block.NumberU64())`.
And then we call startStateMigration after `trieDB.Commit()`.